capstone-ui/src/Components/Feed/Feed.jsx

```diff
@@ -1,87 +1,60 @@
-          "rights": "example.com",
-          "rank": 987,
-          "topic": "education",
-          "country": "US",
-          "language": "en",
-          "authors": ["Jane Doe"],
-          "media": "https://example.com/images/power-of-books.jpg",
-          "is_opinion": false,
-          "twitter_account": "@example",
-          "_score": 9.782,
-          "_id": "23f9j49dsk392rjf93jf02k"
-        }
-      ]);
-    /*const [searchQuery, setSearchQuery] = useState([]);
+    const [articles, setArticles] = useState([]);
+    const [searchQuery, setSearchQuery] = useState([]);
 
     const user = useContext(UserContext);
 
     const currentUser = user.user;
 
+    const [isFetching, setIsFetching] = useState(false);
+    const fetchTimerRef = useRef(null);
+
+    const fetchData = async () => {
+      try {
+        const url = `http://localhost:3001/news?query=${JSON.stringify(searchQuery)}&loggedIn=${loggedin}`;
+
+        const response = await fetch(url, {
+          method: "GET",
+          credentials: "include",
+        });
+
+        const data = await response.json();
+
+        setArticles(data.articles);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
     useEffect(() => {
-        if (loggedin) {
-          const tagsArray = Object.values(currentUser.stances).map(([tag, option]) => tag);
-          setSearchQuery(tagsArray.concat(currentUser.following));
-        }
-      }, [loggedin, user.stances, user.following]);
-    
-    useEffect(() => {
-        const fetchNewsArticles = async () => {
-          try {
-            const url = `http://localhost:3001/news?query=${JSON.stringify(searchQuery)}&loggedIn=${loggedin}`;
-    
-            const response = await fetch(url, {
-                method: "GET",
-                credentials: "include",
-            });
+      if (loggedin) {
+        const tagsArray = Object.values(currentUser.stances).map(([tag, option]) => tag);
+        setSearchQuery(tagsArray.concat(currentUser.following));
+      }
+    }, [loggedin, user.stances, user.following]);
 
-            const data = await response.json();
-    
-            setArticles(data.articles);
-          } catch (error) {
-            console.error(error);
-          }
-        };
-    
-        fetchNewsArticles();
-      }, [searchQuery, loggedin]);*/
+    useEffect(() => {
+      if (isFetching && fetchTimerRef.current) {
```

Comment:
I would take a look at `Fetching with useEffects` to avoid race conditions. https://react.dev/reference/react/useEffect#fetching-data-with-effects